extensions-core/avro-extensions/src/main/java/io/druid/data/input/avro/SchemaRegistryBasedAvroBytesDecoder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.data.input.avro;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.parsers.ParseException;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public class SchemaRegistryBasedAvroBytesDecoder implements AvroBytesDecoder
+{
+  private final SchemaRegistryClient registry;
+
+  @JsonCreator
+  public SchemaRegistryBasedAvroBytesDecoder(
+      @JsonProperty("url") String url,
+      @JsonProperty("capacity") Integer capacity
+  )
+  {
+    int identityMapCapacity = capacity == null ? Integer.MAX_VALUE : capacity;
+    this.registry = new CachedSchemaRegistryClient(url, identityMapCapacity);
+  }
+
+  @Override
+  public GenericRecord parse(ByteBuffer bytes)
+  {
+    bytes.get(); // ignore first \0 byte
+    int id = bytes.getInt(); // extract schema registry id
+    int length = bytes.limit() - 1 - 4;
+    int offset = bytes.position() + bytes.arrayOffset();
+    try {
```

Comment:
schema registry repo contains the avro serializer/deserializer , is it possible to use those instead of us knowing the format of the message?